common/lib/dependabot/dependency_file.rb

```diff
@@ -84,6 +91,8 @@ def initialize(name:, content:, directory: "/", type: "file",
       @vendored_file = vendored_file
       @content_encoding = content_encoding
       @operation = operation
+      @mode = mode
+      raise ArgumentError, "Invalid Git mode: #{mode}" if mode && !VALID_MODES.include?(mode)
```

Comment:
For my understanding, can you share why `nil` is something we should allow? I understand that it is a value that we currently handle.

Reply:
Prior to https://github.com/dependabot/dependabot-core/pull/6141 mode was always nil... well it wasn't even a field. We relied on the PullRequestCreator to set the mode based on the type. 

So now I am returning to that mostly, except it does seem good to allow for overriding this value to fix the executable problem per ecosystem. However adding an error on bad values immediately if one is provided seems like another good thing to do.